File: EASY/falsealarm.cpp
#include <iostream>
#include <vector>

using namespace std;

int func(){
    int n, x, count = 0;
    vector<int> state;
    cin >> n >> x;
    for (int i = 0; i < n; i++){
        int y;
        cin >> y;
        state.push_back(y);
    }
    for(int i = 0; i < n; i++){
        if(state[i] == 0){
            continue;
        }
        else{
            if(i+x >= n){
                cout << "YES" << "\n";
                return 0;
            }
            else{
                for(int z = i+x; z < n; z++){
                    if(state[z] == 1){
                        cout << "NO" << "\n";
                        return 0;
                    }
                    count += 1;
                }
                if (count == n-(i+x)){
                    cout << "YES" << "\n";
                    return 0;
                }
            }
        }
    }
}


int main(){
    int t;
    cin >> t;
    while (t--){
        func();
    }
}

File: EASY/falsealarm.py
def solve():
    t = int(input())
    for _ in range(t):
        doors, time = map(int, input().split())
        state = list(map(str, input().split()))
        count = 0
        for i in state:
            if i == "0":
                count += 1
                if count == doors:
                    print("YES")
            else:
                num = doors - state.index(i) - 1
                if num <= time:
                    print("YES")
                else:
                    print("NO")
                    
if __name__ == "__main__":
    solve()